packages/input-option/src/InputOption/InputOption.tsx

```diff
@@ -36,31 +34,46 @@ export const InputOption = Polymorphic<InputOptionProps>(
     ref,
   ) => {
     const { Component } = usePolymorphic(as);
-    const { theme } = useDarkMode(darkModeProp);
+    const { theme, darkMode } = useDarkMode(darkModeProp);
     return (
-      <Component
-        ref={ref}
-        role="option"
-        aria-selected={highlighted}
-        aria-disabled={disabled}
-        tabIndex={-1}
-        className={cx(
-          inputOptionStyles,
-          inputOptionThemeStyles[theme],
-          {
-            [inputOptionWedge]: showWedge,
-            [inputOptionHoverStyles[theme]]: isInteractive,
-            [inputOptionActiveStyles[theme]]:
-              isInteractive && (selected || highlighted),
-            [inputOptionDisabledStyles[theme]]: disabled,
-            [titleSelectionStyles]: selected,
-          },
-          className,
-        )}
-        {...rest}
+      <InputOptionContext.Provider
+        value={{
+          checked,
+          darkMode,
+          disabled,
+          highlighted,
+        }}
       >
-        {children}
-      </Component>
+        <Component
+          ref={ref}
+          role="option"
```

Comment:
is this correct when it's a menuitem? we might need to fork on that and change some things here..